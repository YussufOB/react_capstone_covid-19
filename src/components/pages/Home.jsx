import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { FiSearch } from 'react-icons/fi';
import { BsArrowRightCircle } from 'react-icons/bs';
import { Link } from 'react-router-dom';
import { fetchCountryData } from '../../redux/Covid';
import Hero from '../Hero';

const Home = () => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchCountryData());
  }, []);

  const countriesData = useSelector((state) => state.countriesData);
  const filteredCountriesData = countriesData.filter((data) => (data.continent === 'Africa'));
  const [postCountry, setCountry] = useState('');
  const searchCountry = (country) => setCountry(country);

  return (
    <>
      <Hero />
      <section>
        <div className="search">
          <FiSearch />
          <input placeholder="search country..." onChange={(e) => searchCountry(e.target.value)} />
        </div>
        <div className="data_display">
          <ul className="country_list">
            {
              filteredCountriesData.filter((value) => value.name
                .toLowerCase().includes(postCountry
                  .toLocaleLowerCase())).map((data) => (
                    <Link
                      to={`/details/${data.name}`}
                      key={data.name}
                      state={data.country === null ? ''
                        : {
                          country: data.name,
                          flag: data.flag,
                          cases: data.cases,
                          tests: data.tests,
                          recovered: data.recovered,
                          active: data.active,
                          critical: data.critical,
                          testPerPeople: data.testPerPeople,
                          casePerPeople: data.casePerPeople,
                          deathPerPeople: data.deathPerPeople,
                          deaths: data.deaths,
                          population: data.population,
                        }}
                    >
                      <div>
                        <div><BsArrowRightCircle /></div>
                        <img src={data.flag} alt={data.country} />
                        <div>
                          <p>{data.name}</p>
                          <p>{data.cases}</p>
                        </div>
                      </div>
                    </Link>
              ))
            }
          </ul>
        </div>
      </section>
    </>
  );
};

export default Home;
