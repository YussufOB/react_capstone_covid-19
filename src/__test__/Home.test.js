import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/ReduxStore';
import Home from '../components/pages/Home';

it('renders properly', () => {
  const home = renderer
    .create(<Provider store={store}><Router><Home /></Router></Provider>)
    .toJSON();
  expect(home).toMatchSnapshot();
});
