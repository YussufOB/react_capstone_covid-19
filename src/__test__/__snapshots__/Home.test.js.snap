// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `
<section
  className="body"
>
  <section
    className="header-cover"
  >
    <div
      className="hero-cover"
    />
    <div
      className="hero"
    >
      <h3>
        COVID-19
      </h3>
      <small>
        Modern Day Most Strong Pandemic
      </small>
      <p>
        "You can stop the spread by wearing your nose mask!!!"
      </p>
    </div>
  </section>
  <div
    className="main-container"
  >
    <span
      className="search-container"
    >
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="11"
          cy="11"
          r="8"
        />
        <line
          x1="21"
          x2="16.65"
          y1="21"
          y2="16.65"
        />
      </svg>
    </span>
    <input
      onChange={[Function]}
      placeholder="locate your country..."
    />
  </div>
  <div
    className="countries-container"
  >
    <ul
      className="countries-list"
    />
  </div>
  <footer>
    <p>
      Â© 2022 BASH | COVID-19 Tracker
    </p>
  </footer>
</section>
`;
